### PR TITLE
Update androidx

### PR DESCRIPTION
**Pull request by vokhuyetOz**
Just migrating to AndroidX

### CODE REVIEW

File: examples/basic/android/app/src/main/java/com/videoplayer/MainApplication.java [hunk 1]
```diff
@@ -24,6 +26,8 @@ public boolean getUseDeveloperSupport() {
     protected List<ReactPackage> getPackages() {
       return Arrays.asList(
         new MainReactPackage(),
+            new ReactVideoPackage(),
+            new ReactVideoPackage(),
```
Comment (jensellfors) [inline examples/basic/android/app/src/main/java/com/videoplayer/MainApplication.java]: Importing and adding `ReactVideoPackage` twice. Is this compiling? @vokhuyetOz

**Comment by vokhuyetOz** [comment]
Oh, it is my mistake, I used react-native link twice

**Comment by vokhuyetOz** [comment]
I will make a commit tomorrow

**Comment by jensellfors** [comment]
Thanks for this @vokhuyetOz. Looks like this forces people to use AndroidX? Or is this backwards compatible? (I'm not too familiar with Android :p )

**Comment by vokhuyetOz** [comment]
Now, people have to use AndroidX whenever upgrade to new build tool version.  As I know, google moves android support responsibilities to new location

**Comment by vokhuyetOz** [comment]
From yesterday, I couldn’t  build android app, the error is can’t import v4 AppCompat

**Comment by jensellfors** [comment]
Have you upgraded to RN 0.60RC? From my understanding this will only be required after upgrading to RN 0.60 https://github.com/react-native-community/discussions-and-proposals/issues/129

**Comment by jensellfors** [comment]
And btw, I installed from your branch and then the android build fails. So it's not backwards compatible.

**Comment by vokhuyetOz** [comment]
No, i still use 0.59

**Comment by vokhuyetOz** [comment]
I just run it in example, i will check it again tomorrow, thank you for your report :)

**Comment by vokhuyetOz** [comment]
@jenshandersson  which error did you have? 
i try and it works.

**Comment by vokhuyetOz** [comment]
make sure that you have updated your gradle.properties and app/build.gradle.
i just have updated read.me file for this change :)

**Comment by kuraydev** [comment]
Waiting for this PR :) Thank you @vokhuyetOz

**Comment by jensellfors** [comment]
@vokhuyetOz after doing your changes I get compile errors in other 3rd party libs... I assume those haven't been upgraded to use AndroidX then? One of them is [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler/issues/647) (that is the error I'm getting as well).

```
> Task :react-native-gesture-handler:compileDebugJavaWithJavac FAILED
/Users/jensa/Developer/jensun/wingsy/RNWingsy/node_modules/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerEvent.java:3: error: package android.support.v4.util does not exist
import android.support.v4.util.Pools;
```

```
> Task :react-native-purchases:compileDebugJavaWithJavac FAILED
/Users/jensa/Developer/jensun/wingsy/RNWingsy/node_modules/react-native-purchases/android/src/main/java/com/reactlibrary/RNPurchasesModule.java:4: error: package android.support.annotation does not exist
import android.support.annotation.Nullable;
```

How do you propose I/we move forward with this? If we would merge this PR then a lot of people would have the same issues, right? (Sorry again for being a bit slow on the Android side)

**Comment by vokhuyetOz** [comment]
a lot of libraries must migrate to AndroidX to fix it

**Comment by vokhuyetOz** [comment]
i am also making a pull request for another library

**Comment by vokhuyetOz** [comment]
i think i should make a PR for react-native-gesture-handler to fix it

**Comment by vokhuyetOz** [comment]
@WrathChaos  could you test this project?

**Comment by kuraydev** [comment]
@vokhuyetOz I already tried it, other libraries also crushes :( We need to update all of them... AndroidX ruined my day..

**Comment by vokhuyetOz** [comment]
you should create a new project and check only this library :)

**Comment by kuraydev** [comment]
@vokhuyetOz  I will try it when I have time :) Thank you for this contribution anyway.

**Comment by jensellfors** [comment]
@vokhuyetOz after speaking to @kelset I now understand that AndroidX upgrade is not required. Since rn-video has a `+` as fallback in the `build.gradle` it will automatically pull in the latest version, and that is AndroidX.  Instead of `+` we need to specify a version. (Anyone with Android knowledge please jump in here if I'm wrong)

Can you please create a new PR with setting explicit versions instead. Upgrading to 28 is fine. But we can't force a AndroidX upgrade of all users right now. See this PR: https://github.com/react-native-community/react-native-device-info/commit/9588763593937d8944e6eb7535521719ee0eeb1e

We will keep this PR as it will be required to move to AndroidX for RN 0.60, but let's wait with that.

**Comment by vokhuyetOz** [comment]
Thank you, at the moment, it is not required, but from 11/2019 developers must update their apps whenever create or update new version. I will take a look at react native device info and make a change

**Comment by vokhuyetOz** [comment]
@jenshandersson  please check out https://github.com/react-native-community/react-native-video/pull/1629 . thank you !

**Comment by n1ru4l** [comment]
react-native 0.60 has now its first stable release. Should merge this asap and publish a new major release.

**Comment by CHaNGeTe** [comment]
This PR will make also possible to update exoplayer to 2.10.x? or totally unrelated?

File: examples/basic/android/build.gradle [hunk 2]
```diff
@@ -15,6 +24,12 @@ buildscript {
 allprojects {
     repositories {
         mavenLocal()
+        maven {
```
Comment (benoitdion) [inline examples/basic/android/build.gradle]: could this be `google()`?

File: examples/basic/android/settings.gradle [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+project(':react-native-video').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-video/android-exoplayer')
 
 // Quick Local Development
 //project(':react-native-video').projectDir = new File(rootProject.projectDir, '../../android')
 //project(':react-native-video-exoplayer').projectDir = new File(rootProject.projectDir, '../../android-exoplayer')
 
-project(':react-native-video').projectDir = new File(rootProject.projectDir, '../../android')
-project(':react-native-video-exoplayer').projectDir = new File(rootProject.projectDir, '../../android-exoplayer')
+// project(':react-native-video').projectDir = new File(rootProject.projectDir, '../../android')
+// project(':react-native-video-exoplayer').projectDir = new File(rootProject.projectDir, '../../android-exoplayer')
```
Comment (benoitdion) [inline examples/basic/android/settings.gradle]: can this be removed?

Comment (vokhuyetOz) [reply examples/basic/android/settings.gradle]: yes,i made a new commit to remove it

File: examples/basic/android/build.gradle (hunk 2 above)
Comment (vokhuyetOz) [reply examples/basic/android/build.gradle]: sorry, my mistake. this could be `google()`

**Comment by CHaNGeTe** [comment]
Can we align with this PR also?
https://github.com/react-native-community/react-native-video/pull/1660

**Comment by vokhuyetOz** [comment]
Yes, of course

**Comment by CHaNGeTe** [comment]
@vokhuyetOz does this change the min version of React required?

**Comment by vokhuyetOz** [comment]
@CHaNGeTe no, it doesn't relate to min version of react required, sorry for my late.

**Review by CHaNGeTe — APPROVED** [review]

**Comment by CHaNGeTe** [comment]
hey @ashnfb @cobarx @n1ru4l @jenshandersson I need another approval for this one

**Review by jensellfors — APPROVED** [review]

**Review by n1ru4l — APPROVED** [review]
Legit for breaking change release (major version number bump)

**Comment by CHaNGeTe** [comment]
What makes it breaking change? (if RN version required does not change)

**Comment by CHaNGeTe** [comment]
ok! clear now!

**Comment by CHaNGeTe** [comment]
@vokhuyetOz I am preparing the merge. Do you know the minimum required gradle version for this to work?

**Comment by agungkes** [comment]
Aha it's good to know this PR will merge 
Thanks you @CHaNGeTe

**Comment by karltaylor** [comment]
Woohoo! 🎉

File: android-exoplayer/build.gradle [hunk 4]
```diff
@@ -28,9 +28,12 @@ dependencies {
     }
 
     // All support libs must use the same version
-    implementation "com.android.support:support-annotations:${safeExtGet('supportLibVersion', '+')}"
-    implementation "com.android.support:support-compat:${safeExtGet('supportLibVersion', '+')}"
-    implementation "com.android.support:support-media-compat:${safeExtGet('supportLibVersion', '+')}"
+    // implementation "com.android.support:support-annotations:${safeExtGet('supportLibVersion', '+')}"
```
Comment (benoitdion) [inline android-exoplayer/build.gradle]: can this be removed instead of commented out?

File: examples/basic/android/app/build.gradle [hunk 5]
```diff
@@ -108,6 +111,7 @@ android {
         release {
             signingConfig signingConfigs.debug
             minifyEnabled enableProguardInReleaseBuilds
+            matchingFallbacks = ['release', 'debug']
```
Comment (benoitdion) [inline examples/basic/android/app/build.gradle]: why is this needed?

Comment (vokhuyetOz) [reply examples/basic/android/app/build.gradle]: i don't remember exactly why i add it. i will remove it in the next commit :D

**Comment by vokhuyetOz** [comment]
> @vokhuyetOz I am preparing the merge. Do you know the minimum required gradle version for this to work?

sorry, i don't find any information about the minimum required gradle version. But i find 
Screen Shot 2019-07-24 at 9 08 05 AM

Hope it helps you!
